packages/lodestar/src/api/impl/beacon/pool/pool.ts

```diff
@@ -71,22 +74,25 @@ export class BeaconPoolApi implements IBeaconPoolApi {
   }
 
   public async submitAttesterSlashing(slashing: AttesterSlashing): Promise<void> {
-    await this.db.attesterSlashing.add(slashing);
+    await validateGossipAttesterSlashing(this.config, this.chain, this.db, slashing);
+    await Promise.all([this.network.gossip.publishAttesterSlashing(slashing), this.db.attesterSlashing.add(slashing)]);
   }
 
   public async getProposerSlashings(): Promise<ProposerSlashing[]> {
     return this.db.proposerSlashing.values();
   }
 
   public async submitProposerSlashing(slashing: ProposerSlashing): Promise<void> {
-    await this.db.proposerSlashing.add(slashing);
+    await validateGossipProposerSlashing(this.config, this.chain, this.db, slashing);
+    await Promise.all([this.network.gossip.publishProposerSlashing(slashing), this.db.proposerSlashing.add(slashing)]);
```

Comment:
I think we may want to be 'greedy' with the slashings we receive and not publish them in gossip.

Reply:
what's the reason to do that?  From the spec: `Submits ProposerSlashing object to node's pool and if passes validation node MUST broadcast it to network.`

Reply:
so i guess this issue is tracking this now?
https://github.com/ChainSafe/lodestar/issues/1956